src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -266,31 +266,72 @@ trait TypeOps { this: Context => // TODO: Make standalone object.
         val accu1 = if (accu exists (_ derivesFrom c)) accu else c :: accu
         if (cs == c.baseClasses) accu1 else dominators(rest, accu1)
     }
-    if (ctx.featureEnabled(defn.LanguageModuleClass, nme.keepUnions)) tp
-    else tp match {
-      case tp: OrType =>
-        def isClassRef(tp: Type): Boolean = tp match {
-          case tp: TypeRef => tp.symbol.isClass
-          case tp: RefinedType => isClassRef(tp.parent)
-          case _ => false
-        }
-        def next(tp: TypeProxy) = tp.underlying match {
-          case TypeBounds(_, hi) => hi
-          case nx => nx
-        }
-        tp.tp1 match {
-          case tp1: TypeProxy if !isClassRef(tp1) =>
-            approximateUnion(next(tp1) | tp.tp2)
-          case _ =>
-            tp.tp2 match {
-              case tp2: TypeProxy if !isClassRef(tp2) =>
-                approximateUnion(tp.tp1 | next(tp2))
+    def approximateOr(tp1: Type, tp2: Type)(implicit ctx: Context): Type = {
```

Comment:
Indeed. That was a leftover of a previous version where something got added to the context.